app/sheet/read-data.c

```diff
@@ -207,6 +207,8 @@ static int read_data(struct zsvsheet_ui_buffer **uibufferp,   // a new zsvsheet_
   pthread_mutex_lock(&uibuff->mutex);
   char need_index = !uibuff->index_started && !uibuff->write_in_progress;
   char *old_ui_status = uibuff->status;
+  if (need_index)
+    uibuff->index_started = 1;
```

Comment:
Just wondering; I don't think there should be a race condition here because the UI buffer should not have a worker attached at this point and this is executing in the main thread?

Reply:
You're probably right but this way we don't have to think about it, or worry about some change in the future adding a worker before this block executes